### ISSUE TITLE
Error: Cannot find module `remark-preset-lint-recommended`

### ISSUE THREAD

**Issue by nvuillam** [remarkjs/remark-lint#322]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

v15.0.1

### Steps to reproduce

.remarkrc is the following

```json
{
  "plugins": [
    "remark-preset-lint-recommended"
  ]
}
```

See MegaLinter failing job

https://github.com/oxsecurity/megalinter/actions/runs/8862094296/job/24334663734?pr=3516

It used to work in previous versions

### Expected behavior

To work ^^

### Actual behavior

The following error is triggered

    Error: Cannot find module `remark-preset-lint-recommended`
    at Function.<anonymous> (file:///node-deps/node_modules/unified-engine/lib/configuration.js:432:15)

### Runtime

Node v17

### Package manager

npm 8

### OS

Linux

### Build and bundle tools

Other (please specify in steps to reproduce)

**Comment by ChristianMurphy** [comment]
Welcome @nvuillam! 👋 
Sorry you ran into a spot of trouble.

`remark-cli` and `remark-preset-lint-recommended` are working fine together https://stackblitz.com/edit/node-rgxynu
I'm not sure what version 15.0.1 refers to neither of those packages have release a version 15 yet.

Feel free to share a minimal reproducible example of the issue you are seeing, ideally in a sandbox.
From the brief look at megalinter, there looks like there is a lot of abstraction wrapping these in the linter, and it feels like the bug may be in that abstraction.

**Comment by nvuillam** [comment]
@ChristianMurphy remark-lint is disabled in MegaLinter since... July 2022, maybe you have a hint of a major change that happened at that time ?

Maybe the way to find remark-preset-lint-recommended package if remarklink-cli is installed globally ?

It is installed the following way (but not in the projet, in a MegaLinter docker image custom folder): ` npm --no-cache install --ignore-scripts --omit=dev remark-cli remark-preset-lint-recommended`

Here is how to version is retrieved, and the result

```sh
Linter version command: ['/node-deps/node_modules/.bin/remark', '--version']
Linter version result: 0 remark: 15.0.1, remark-cli: 12.0.0
```

**Comment by ChristianMurphy** [comment]
> maybe you have a hint of a major change that happened at that time?

A lot have happened.
There are changelogs with migration which document all of those.
I'd encourage reading them 🙂
* https://github.com/remarkjs/remark/releases
* https://github.com/remarkjs/remark-lint/releases

> `/node-deps/node_modules/.bin/remark`

This stands put to me, crossing folders to get at node modules could be messing with module resolution.

Try using an `npm` script or `npx` to run remark.

**Comment by remcohaszing** [comment]
> It is installed the following way (but not in the projet, in a MegaLinter docker image custom folder): ` npm --no-cache install --ignore-scripts --omit=dev remark-cli remark-preset-lint-recommended`

So it’s neither installed globally, nor locally in the project. Instead, it is installed locally in another project. This is quite an edge case. `remark-preset-lint-recommended` is looked up both globally and from the current project. You’re suggesting it should additionally be looked up locally in the project where `remark-cli` is installed.

I’m not really for or against this personally. MegaLinter is better off installing packages globally by passing the **`--global`** flag to npm. I feel like the argument for supporting this would be stronger if MegaLinter pinned their dependencies using a lockfile.

Regardless of all this, I strongly recommend to specify the major version of packages to install.

**Comment by wooorm** [comment]
If you install things somewhere, say, `path/to/remark-whatever-package`, you can use that:

```json
{
  "plugins": [
    "path/to/remark-whatever-package/index.js"
  ]
}
```

Or, install things globally.
Or, install things locally.

**Comment by nvuillam** [comment]
Thanks for your responses :)
